src/xlsx.js

```diff
@@ -0,0 +1,98 @@
+export class ExcelWorkbook {
+  constructor(workbook) {
+    Object.defineProperty(this, "_", {value: workbook});
+  }
+  sheetNames() {
+    return this._.worksheets.map((sheet) => sheet.name);
+  }
+  sheet(name, {range, headers = false} = {}) {
+    const sheet = this._.getWorksheet(
+      typeof name === "number" ? this.sheetNames()[name] : name + ""
+    );
+    if (!sheet) throw new Error(`Sheet not found: ${name}`);
+    return extract(sheet, {range, headers});
+  }
+}
+
+function extract(sheet, {range, headers}) {
+  let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
+  const headerRow = headers && sheet._rows[r0++];
+  const seen = new Set();
+  const names = [];
+  function name(n) {
+    if (!names[n]) {
+      let name = (headerRow ? valueOf(headerRow._cells[n]) : AA(n)) || AA(n);
+      while (seen.has(name)) name += "_";
+      seen.add((names[n] = name));
+    }
+    return names[n];
+  }
+
+  const output = new Array(r1 - r0 + 1).fill({});
+  for (let r = r0; r <= r1; r++) {
+    const _row = sheet._rows[r];
+    if (!_row || !_row.hasValues) continue;
+    const row = (output[r - r0] = {});
+    for (let c = c0; c <= c1; c++) {
+      const value = valueOf(_row._cells[c]);
+      if (value !== null && value !== undefined) row[name(c)] = value;
+    }
+  }
+
+  output.columns = names.filter(() => true);
+  return output;
+}
+
+function valueOf(cell) {
+  if (!cell) return;
+  const {value} = cell;
+  if (value && typeof value === "object") {
+    if (value.formula) return value.result;
+    if (value.richText) return value.richText.map((d) => d.text).join("");
+    if (value.text && value.hyperlink)
+      return `<a href="${value.hyperlink}">${value.text}</a>`;
```

Comment:
might hyperlink contain chars that should be escaped (like a double quote)? or text contain a `</a>`?

Reply:
not really sure how to represent HTML here. should I reuse the html tagged template literal?

Reply:
properly escaped the string. debated bringing in [escape-html](https://www.npmjs.com/package/escape-html), but wrote my own function. 🤔